### PR TITLE
Frontend integration

### PR DESCRIPTION
**Pull request by rasdy**
This PR adds frontend support for GET-ing content for the intro, task assignment, and inital reflection pages.

Also, getInitReflectPage() in queries.js has been changed so that it additionally returns the body text and the prompt IDs for each prompt.

The other get**X**ReflectPage() functions in queries.js have _not_ been changed, but they should be.

The pages added to the scenarios in insert_example_data.sql are mis-ordered, so for the sake of the last mini-demo (at first I didn't realize the editor frontend couldn't add pages other than the intro), for scenario 1 I set the ORDER of the initial reflection to 3, and I changed the conversation page (previously ORDER 3) to a PLAIN page with text that says it's a task assignment.

So basically the pages are still out of order (other than scenario 1's first three pages).

### CODE REVIEW

File: goon-universe/queries.js [hunk 1]
```diff
@@ -540,12 +540,18 @@ function getStakeholderDescriptions(scenarioID){
...
 function getInitReflectPage(scenarioID, callback){
-    let thisQuery= 'select prompt.prompt from pages, prompt where pages.id = prompt.page_id and pages.order = '+ INITIAL_REFLECTION +' and scenario_id = $1'
+    let thisQuery= 'select pages.body_text, prompt.prompt, prompt.prompt_num from pages, prompt where pages.id = prompt.page_id and pages.order = '+ INITIAL_REFLECTION +' and scenario_id = $1'
     pool.query(thisQuery, [scenarioID], (error,results) => {
         if (error) {
             throw error
         }
-        callback(results.rows)
+        let response = {}
+        response.prompts = results.rows.map(row => ({
```
Comment (rlee287) [inline goon-universe/queries.js]: Pinging @miesoz to make sure he is aware of the format change here

**Comment by rlee287** [comment]
The changes to `queries.js` do a bit more manipulation with the query results than I did, so I will add that to my copy and do the same with the other related functions.

With relation to the example data, we got rid of the PLAIN page type in favor of having a body text, so a PLAIN page type is now incorrect. (This may not have been communicated properly to you so I'm repeating it here again).

I'm assuming that Goon Universe has been made aware of the change in callback format, so I'll propagate the changes to the other page functions. (If they do not want this then we can easily roll back that part.)

**Comment by rlee287** [comment]
Everything on `queries.js` should be properly fixed up now.

File: goon-universe/queries.js (hunk 1 above)
Comment (kungoyal) [reply goon-universe/queries.js]: It looks good to me. I've tested it this way and it seems to work fine

File: goon-universe/queries.js [hunk 2]
```diff
@@ -539,33 +539,52 @@ function getStakeholderDescriptions(scenarioID){
...
+            text: row.prompt,
+            id: row.prompt_num
+        }))
+        response.body_text = results.rows[0].body_text
```
Comment (kungoyal) [inline goon-universe/queries.js]: The code works fine in most cases. However, when a scenario id that does not exist in the database is passed, the code crashes with a 'cannot read property body_text of undefined'. This needs to fixed. I think a simple if(results.length == 0) check will solve the problem

File: goon-universe/queries.js (hunk 1 above)
Comment (miesoz) [reply goon-universe/queries.js]: Looked over it with Kunal, seems good.

**Review by miesoz — DISMISSED** [review]

File: goon-universe/queries.js (hunk 2 above)
Comment (kungoyal) [reply goon-universe/queries.js]: Returning(callback) an empty object if that condition is true will be the best thing to do since it will work with the API without having to change anything

**Comment by rlee287** [comment]
@kungoyal When you call `get*ReflectPage`, your callback function needs to change: you need to replace `result.length == 0` with something that uses the actual object that gets returned (an empty `{}` if the ID was invalid or something with `body_text` and `prompts` fields if the ID was valid).

**Review by rlee287 — CHANGES_REQUESTED** [review]
Not ready to merge; see comment I just left

**Review by rlee287 — DISMISSED** [review]
Seems like it is fixed to me; but @rasdy please test locally before merging this in

**Review by miesoz — DISMISSED** [review]

**Comment by rlee287** [comment]
Next steps from integration meeting today:

- Implement for loop to loop over responses when inserting/getting
- Merge this in
- Merge in #201 after this PR

**Review by rlee287 — DISMISSED** [review]

**Comment by rasdy** [comment]
Seems to be mostly working, but I found a couple issues.

 - If the student or scenario ID doesn't exist in the db, a PUT initialReflection request results in an unhandled promise rejection from `addReflectResponse()`
 - The response I get from calling `GET /scenarios/initalReflection/response` with `studentID: 2, scenarioID: 1` is confusing. I get prompts 1 and 2 for scenario 1, but I also get prompt 1 for scenario 2?
Calling `PUT /scenarios/initialReflection` with `studentID: 2, scenarioID: 1, data: {1: "new response"}`, according to `GET /scenarios/initalReflection/response`, overwrites prompt 2 instead of 1.
 - Calling `GET /scenarios/initalReflection/response` with the wrong studentID or scenarioID results in a 200 OK response containing an empty list

**Comment by rlee287** [comment]
For the first point: in `addXReflectResponse`, what should we put in the callback, and how should we handle callbacks when the student or scenario doesn't exist? (We have something right now but I'm asking again to make sure we are all on the same page here.)

**Review by miesoz — DISMISSED** [review]

**Review by miesoz — COMMENTED** [review]
Did not mean to approve, I just saw the merge conflicts.

**Review by rlee287 — CHANGES_REQUESTED** [review]
Merge conflicts are not a reason to deny approval for a PR, as those happen periodically (and we can simply talk out how to resolve those when they occur.) However, there are other issues remaining that @rasdy pointed out earlier, so I'll review this as "request changes" until those get fixed.

**Comment by rlee287** [comment]
Echoing messages from Track2 slack channel:

- First item should be fixed now (following communication with Goon Universe)
- Second item GET works correctly now; unable to reproduce the PUT issue
- Third item: code that handles this was explicitly commented out for an unknown reason

**Review by rlee287 — DISMISSED** [review]
Bullet points should be fixed now

**Review by rlee287 — APPROVED** [review]
